projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+    "bufio"
+    "fmt"
+    "net/http"
+	"os"
+	"time"
+	"strconv"
+	"errors"
+	"io"
+)
+
+func main() {
+    resp, err := http.Get("http://localhost:8080")
+
+   if err != nil {
+		if err != nil {
+		if errors.Is(err, io.EOF) {
+			fmt.Println("The server closed the connection unexpectedly. Please try again later.")
+		} else {
+			fmt.Println("Error:", err)
+		}
+		os.Exit(1)
+	}
+	}
+    defer resp.Body.Close()
+
+    fmt.Println("Response status:", resp.Status)
+
+    scanner := bufio.NewScanner(resp.Body)
+    for i := 0; scanner.Scan() && i < 10; i++ {
+        fmt.Println(scanner.Text())
+    }
+
+    if err := scanner.Err(); err != nil {
+        panic(err)
+    }
+	if resp.StatusCode == http.StatusTooManyRequests {
+        handleTooManyRequests(resp)
+    }
+}
+
+func handleTooManyRequests(resp *http.Response) {
+    retryAfterHeader := resp.Header.Get("Retry-After")
+    if retryAfterHeader != "" {
+        retryAfterSeconds, err := strconv.Atoi(retryAfterHeader)
```

Comment:
This pattern of "try to do one thing, if it worked return, if not, try another" is really nice - it makes it really clear what's going on, and which cases could happen. Good job!